besticon/besticon_test.go

```diff
@@ -13,214 +13,203 @@ import (
-	}
-	assertEquals(t, expectedImages, actualImages)
-
-	actualImage := finder.IconInSizeRange(SizeRange{20, 80, 500})
-	expectedImage := &Icon{URL: "http://info.kicktipp.de/assets/img/jar_cb333387130/assets/img/logos/apple-touch-icon-57x57-precomposed.png", Width: 57, Height: 57, Format: "png", Bytes: 1535, Sha1sum: "79aae9e0df7d52ed50ac47c1dd4bd16e2ddf8b4a"}
-	assertEquals(t, expectedImage, actualImage)
+//
+// Big list of tests for IconFinder.FetchIcons. Responses are cached on disk
+// using our VCR file format. No network connectivity is required unless you are
+// adding more tests.
+//
+
+type testFetch struct {
+	url   string
+	icons []testFetchIcon
 }
 
-func TestDaringfireball(t *testing.T) {
-	actualImages, finder, err := fetchIconsWithVCR("daringfireball.net.vcr", "http://daringfireball.net")
-	assertEquals(t, nil, err)
-
-	expectedImages := []Icon{
-		{URL: "http://daringfireball.net/graphics/apple-touch-icon.png", Width: 314, Height: 314, Format: "png", Bytes: 8642, Error: error(nil), Sha1sum: "f47cf7cf13ec1a74049d99d9f1565354b5b20317"},
-		{URL: "http://daringfireball.net/favicon.ico", Width: 32, Height: 32, Format: "ico", Bytes: 6518, Error: error(nil), Sha1sum: "c066c70aa1dd2b4347d3095592aac28b55e85d04"},
-		{URL: "http://daringfireball.net/graphics/favicon.ico?v=005", Width: 32, Height: 32, Format: "ico", Bytes: 6518, Error: error(nil), Sha1sum: "c066c70aa1dd2b4347d3095592aac28b55e85d04"},
-	}
-
-	assertEquals(t, expectedImages, actualImages)
-
-	actualImage := finder.IconInSizeRange(SizeRange{20, 80, 500})
-	expectedImage := &Icon{URL: "http://daringfireball.net/graphics/apple-touch-icon.png", Width: 314, Height: 314, Format: "png", Bytes: 8642, Error: error(nil), Sha1sum: "f47cf7cf13ec1a74049d99d9f1565354b5b20317"}
-	assertEquals(t, expectedImage, actualImage)
+type testFetchIcon struct {
+	url    string
+	width  int
+	format string
 }
 
-func TestAwsAmazonChangingBaseURL(t *testing.T) {
-	actualImages, _, err := fetchIconsWithVCR("aws.amazon.vcr", "http://aws.amazon.com")
-	assertEquals(t, nil, err)
-	expectedImages := []Icon{
-		{URL: "http://a0.awsstatic.com/main/images/site/touch-icon-ipad-144-precomposed.png", Width: 144, Height: 144, Format: "png", Bytes: 3944, Sha1sum: "225817df40ff11d083c282d08b49a5ed50fd0f2d"},
-		{URL: "http://a0.awsstatic.com/main/images/site/touch-icon-iphone-114-precomposed.png", Width: 114, Height: 114, Format: "png", Bytes: 3081, Sha1sum: "58aabb2a99fcb283710fd200c9feed69c015a29e"},
-		{URL: "http://a0.awsstatic.com/main/images/site/favicon.ico", Width: 16, Height: 16, Format: "ico", Bytes: 1150, Sha1sum: "a64f3616e3671b835f8b208b7339518d8b386a08"},
-		{URL: "http://aws.amazon.com/favicon.ico", Width: 16, Height: 16, Format: "ico", Bytes: 1150, Sha1sum: "a64f3616e3671b835f8b208b7339518d8b386a08"},
+func TestFetchIcons(t *testing.T) {
+	tests := []testFetch{
+		// alibaba - base tag without scheme
```

Comment:
Maybe even worthwhile to treat this as data by adding  it as e.g. `testName` to `testFetch`?

Reply:
Not a bad idea. Only reason I didn't was because I was trying to keep that first line a reasonable length. I'll poke around a bit more for the next PR.

Reply:
Sure, just a thought, judge for yourself in we Gaia anything from it :–)